analyzer/codechecker_analyzer/analyzer_context.py

```diff
@@ -118,24 +213,47 @@ def __iter__(self):
     def __len__(self):
         return len(self.store)
 
+    def by_guideline(self, guideline):
+        """
+        Return checkers belonging to a specific guideline. A checker belongs to
+        a guideline if it reports on at least one guideline rule.
+        """
+        result = []
+        for checker, guidelines in self.store['mapping'].items():
+            if guideline in guidelines and len(guidelines[guideline]) != 0:
```

Comment:
`len(guidelines[guideline])` means the same as `len(guidelines[guideline]) != 0` in conditions but the first one is much shorter:

```suggestion
            if guideline in guidelines and len(guidelines[guideline]):
```

Reply:
It is my personal opinion, but I don't like relying on int->bool conversion in any languages if the integer is describing a length, count, etc. of something. In my eyes this is the same as writing `!strcmp(a, b)` for checking equality in C, I use `strcmp(a, b) == 0` instead. It is quite unfortunate that in Python this is the standard way of checking emptiness. But you know what, I offer a compromise. What about this? :)
```suggestion
            if guideline in guidelines and guidelines[guideline]:
```